lib/ansible/modules/cloud/azure/azure_rm_servers.py

```diff
@@ -0,0 +1,279 @@
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group_name = None
+        self.server_name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.resource_group = None
+        self.state = None
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            else:
+                self.parameters[key] = kwargs[key]
+
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group_name)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group_name))
+
+        response = self.get_servers()
+
+        if not response:
+            self.log("Servers instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Servers instance already exists")
+            if self.state == 'absent':
+                self.delete_servers()
+                self.results['changed'] = True
+                self.log("Servers instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if Servers instance has to be deleted or may be updated")
+                to_be_updated = True
+                if to_be_updated:
+                    self.log('Deleting Servers instance before update')
+                    if not self.check_mode:
+                        self.delete_servers()
+
+        if self.state == 'present':
+
+            self.log("Need to Create / Update the Servers instance")
+
+            if self.check_mode:
+                return self.results
+
+            if to_be_updated:
+                self.results['state'] = self.create_update_servers()
+                self.results['changed'] = True
+            else:
+                self.results['state'] = response
+
+            self.log("Creation / Update done")
+
+        return self.results
+
+    def create_update_servers(self):
```

Comment:
server

Reply:
i think i will just change it to create_or_update and will drop "servers" as i don't know whether name is plural or singular